examples/memory_dataset.py

```diff
@@ -0,0 +1,55 @@
+# Copyright pydicom authors 2019. See LICENSE file for details
+"""
+======================================
+Show working with memory-based dataset
+======================================
+
+Show how to write a DICOM dataset into a byte array and to read
+it back from a byte array. This can be helpful for example if working with
+datasets saved as blobs in a database.
+
+"""
+
+from __future__ import print_function
+
+from io import BytesIO
+
+from pydicom import dcmread, dcmwrite
+from pydicom.filebase import DicomFileLike
+
+print(__doc__)
+
+usage = "Usage: python memory_dataset.py dicom_filename"
+
+
+def write_dataset_to_bytes(filepath):
+    # read the dataset from a file
```

Comment:
A nit-pick, but given name 'write_dataset...' shouldn't it be passed a dataset?  I think it also makes the example closer to real-world use.